src/components/Selector/index.tsx

```diff
@@ -305,7 +305,7 @@ export const WatchProviderSelector = ({
 
   useEffect(() => {
     onChange(watchRegion, activeProvider);
-  }, [activeProvider, watchRegion, onChange]);
+  }, [activeProvider, watchRegion]);
```

Comment:
This probably has a linter warning above it. We should add a comment disabling eslint on this line and explain why we disabled it.

Reply:
Added.